doc/specs/application-auth-token.md

```diff
@@ -20,26 +45,82 @@ A semver string specifying the spec version under which this AAT needs to be int
 
 > type: `string`
 
-The application will sign a hash of the `message` property within this token with the specified `appPubKey` and
-corresponding private key.
+The `Application` will sign a hash of a marshalled AAT structure where all the fields
+are populated with the exception of the `Signature` itself. The resultant `Signature`
+is the one that's populated in the `AAT`.
 
 ### applicationPublicKey
 
 > type: `string`
 
-The hexadecimal publicKey of the Application
+The hexadecimal `publicKey` of the `Application`
 
 ### clientPublicKey
 
 > type: `string`
 
-Required for signature verification, the hexadecimal public of each individual client allowing for granular control of
-who can use the AAT
+Required for signature verification.
+
+Also known as the `gatewayPublicKey`.
+
+The hexadecimal public key allowing granular control of who can use the `AAT`.
 
 ## ECDSA ed25519 Signature Scheme
 
-The protocol wide ed25519 ECDSA will be used for any signatures and verifications that are used within this
-specification.
+The protocol wide ed25519 ECDSA will be used for any signatures and verifications
+that are used within this specification.
+
+## AAT Generation
+
+When generating a new AAT, the owner of the `Application` private key can choose
+which `Client` public key is used to sign Relays. More explicitly, this can be
+decomposed into two options:
+
+1. **Application Key != Client Key ** - Recommended for all Gateways.
+2. **Application Key == Client Key** - Recommended for independent Applications that are not Gateways.
```

Comment:
This part is still confusing to me.  Having "Recommended for all Gateways" and "Recommended for independent Applications that are not Gateways" in the same list doesn't sound right.  The first item says "for gateways" while the second item says "for Applications", but these are two different concepts.  Application is not Gateway.

I think it's fair to say we create AAT for gateways because a gateway (aka client, relayer) is a program to use AAT.  Then the second case confuses me.  Do you mean there is a situation where we can use AAT without any gateways?  Can you elaborate that case a little bit more?

Reply:
You should watch my talk! https://www.youtube.com/watch?v=jVW-3lRzVT0&t=1s

Docs also updated.

Reply:
I added this:

```
_Note: Please note that option 2 is only shown for theoretical purposes. Due to
the scaling limitations of Morse (v0) implemented in `pocket-core`, Gateways
have to be used. The original design / intention of Pocket, and how Shannon (v1)
will operate is support both modes._
```

If this is not enough and you want me to write out the whole story, I'd rather just delete these updates ot the docs altogether.

Lmk what you prefer.

Reply:
Hmm, looks like we're not yet on the same page.  Before just stepping back, let's discuss it more.  I think it's important.  If I miss something, I want to know it.

First of all, I watched https://www.youtube.com/watch?v=jVW-3lRzVT0.  That was a great talk!

I watched "Sovereign Application" part at around 15:00 carefully and if I understand that part correctly, the Sovereign Application scenario still requires a gateway, right?  As you said "they talk to the suppliers themselves" in the talk, they have their own program to submit relays, which is a gateway.

I guess you're calling only a delegated client "gateway".  If so, "gateway" is just one usecase of "client" and we can't say "client, aka gateway".  Whether it's delegated or not, a client is a program to submit relays.  Actually I personally prefer calling it "relayer".

Anyway, the question is if "Sovereign Application" operator doesn't delegate the relayer part to anyone and opereates it themselves, is it recommended to use the same key for the Gateway and App?  My answer is no.

To me, this question is almost the same as "can we use the same password for multiple services, like Amazon.com account and Google account?".  The benefit from doing that is we can reduce the number of passwords to be managed, but of course the answer is "we can, but it's not recommended because it sacrifices security".

I think Application/Gateway has the same story.

In whatever scenario, a gateway program needs to access client private key to send a request.  It needs to load the key in a raw format at some point.  Access to application private key, on the other hand, is not required if they precreate an AAT.  And if they do so, even if their server is hacked, application private key is safe.  Good.  If they use the same key for Client and Application, application private key is leaked when their client is hacked, and the staked tokens will be at stake really.

Maybe I'm getting paranoid to this.  Probably the server will not be hacked.  Managing less private keys is more beneficial.  But if you *recommend* something, you need to clarify pros and cons, and explain why.  Otherwise people cannot choose which fits their system "Application Key != Client Key" or "Application Key == Client Key" .

Reply:
The goal is to deprecate `pocket-core` and focus on `poktroll`, so I don't want to make a big deal out of this. This was a "side quest".

How about I just delete this part of the documentation to avoid confusion and paranoia?
👍?

Reply:
We discussed in Discord and agreed:
- We should not bring Shannon concepts to this repo
- Client in v0 is not Gateway in Shannon
- Client pubkey may be the same as App pubkey, but it's not recommended in v0